eos/utils/szego-polynomial_TEST.cc

```diff
@@ -37,45 +37,56 @@ class SzegoPolynomialTest :
-					TEST_CHECK_RELATIVE_ERROR(p4, +5.166081,           1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p5, -9.464102,           1.0e-5);
-				}
+                    TEST_CHECK_RELATIVE_ERROR(p0, +0.6351351032391984, 1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p1, -0.847745,           1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p2, +1.54489,            1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p3, -2.82388,            1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p4, +5.166081,           1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p5, -9.464102,           1.0e-5);
+                }
 
-				{
-					const auto [p0, p1, p2, p3, p4, p5] = p(+0.0);
+                {
+                    const auto [p0, p1, p2, p3, p4, p5] = p(+0.0);
 
-					TEST_CHECK_RELATIVE_ERROR(p0, +0.6351351032391984, 1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p1, -0.749503,           1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p2, +1.304458,           1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p3, -2.237009,           1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p4, +3.834085,           1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p5, -6.577731,           1.0e-5);
-				}
+                    TEST_CHECK_RELATIVE_ERROR(p0, +0.6351351032391984, 1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p1, -0.749503,           1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p2, +1.304458,           1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p3, -2.237009,           1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p4, +3.834085,           1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p5, -6.577731,           1.0e-5);
+                }
 
-				{
-					const auto [p0, p1, p2, p3, p4, p5] = p(+0.1);
+                {
+                    const auto [p0, p1, p2, p3, p4, p5] = p(+0.1);
 
-					TEST_CHECK_RELATIVE_ERROR(p0, +0.6351351032391984, 1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p1, -0.651261,           1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p2, +1.09668,            1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p3, -1.76188,            1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p4, +2.82970,            1.0e-5);
-					TEST_CHECK_RELATIVE_ERROR(p5, -4.54691,            1.0e-5);
-				}
-			}
+                    TEST_CHECK_RELATIVE_ERROR(p0, +0.6351351032391984, 1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p1, -0.651261,           1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p2, +1.09668,            1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p3, -1.76188,            1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p4, +2.82970,            1.0e-5);
+                    TEST_CHECK_RELATIVE_ERROR(p5, -4.54691,            1.0e-5);
+                }
+
+                {
+                    const auto [p0, p1, p2, p3, p4, p5] = p(complex<double>(0.4, 0.916515139));
+
+                    TEST_CHECK_RELATIVE_ERROR(p0, +0.6351351032391984, 1.0e-5);
```

Comment:
Ah, I should have seen this earlier. You need to check the relative error for ``std::real(p0)`` only. macOS complains that ``p0`` cannot be converted to double when passing to ``std::fabs(const double &)``, while the others apparently have an overload for ``std::fabs(const complex<double> &)``. This puzzles me. I think this should fail!

Reply:
```
  template<typename _Tp>
    inline _Tp
    /// fabs(__z) [8.1.8].
    //  Effects:  Behaves the same as C99 function cabs, defined
    //            in subclause 7.3.8.1.
    fabs(const std::complex<_Tp>& __z)
    { return std::abs(__z); }
```
macOS is on the wrong side of the standard here. As of TR1 the C++ standard demands that ``fabs`` works for a complex number, to ensure compatibility with C99.

Reply:
Ok makes sense, thanks :)